File: src/main/java/toker/warbandscripts/panel/entity/Inventory.java
package toker.warbandscripts.panel.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import org.springframework.data.jpa.repository.EntityGraph;

import javax.persistence.*;
import java.util.Collection;
import java.util.Objects;

@Entity
@Table(name = "inventory")
public class Inventory {
    private Integer id;
    private Integer size;
    private Player player;
    private Collection<InventorySlot> slots;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Basic
    @Column(name = "size", nullable = false)
    public Integer getSize() {
        return size;
    }

    public void setSize(Integer size) {
        this.size = size;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Inventory inventory = (Inventory) o;
        return Objects.equals(id, inventory.id) &&
                Objects.equals(size, inventory.size);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, size);
    }

    @ManyToOne
    @JoinColumn(name = "player_id", referencedColumnName = "id", nullable = false)
    @JsonIgnore
    public Player getPlayer() {
        return player;
    }

    public void setPlayer(Player playerByPlayerId) {
        this.player = playerByPlayerId;
    }

    @OneToMany(mappedBy = "inventory", cascade = {CascadeType.MERGE, CascadeType.PERSIST})
    public Collection<InventorySlot> getSlots() {
        return slots;
    }

    public void setSlots(Collection<InventorySlot> inventorySlotsById) {
        this.slots = inventorySlotsById;
    }
}

File: src/main/java/toker/panel/entity/Inventory.kt
package toker.panel.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "inventory")
class Inventory(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", nullable = false)
        var id: Int? = null,

        @Column(name = "size", nullable = false)
        var size: Int? = null,

        @ManyToOne
        @JoinColumn(name = "player_id", referencedColumnName = "id", nullable = false)
        @JsonIgnore
        var player: Player? = null
) {
    @OneToMany(mappedBy = "inventory", cascade = [CascadeType.MERGE, CascadeType.PERSIST])
    var slots: MutableSet<InventorySlot>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Inventory
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}